app/domain/authentication/authn_k8s/extract_k8s_resource_restrictions.rb

```diff
@@ -0,0 +1,66 @@
+require 'command_class'
+
+module Authentication
+  module AuthnK8s
+
+    ExtractK8sResourceRestrictions = CommandClass.new(
+      dependencies: {
+        extract_resource_restrictions: Authentication::ResourceRestrictions::ExtractResourceRestrictions.new,
+        resource_restrictions_class:   Authentication::ResourceRestrictions::ResourceRestrictions,
+        logger:                        Rails.logger
+      },
+      inputs:   %i(authenticator_name service_id role_name account)
+    ) do
+
+      def call
+        extract_resource_restrictions_from_annotations_or_host_id
+      end
+
+      private
+
+      def extract_resource_restrictions_from_annotations_or_host_id
```

Comment:
please add a comment that we first check if there are restrictions in annotations and only then in the id. If there are restrictions in both places we take only the annotations

Reply:
done